pkg/storage/article_storage.go

```diff
@@ -0,0 +1,19 @@
+package storage
+
+import (
+	"news-aggregator/pkg/model"
+	"time"
+)
+
+//go:generate mockgen -destination=mocks/mock_article.go -package=mocks news-aggregator/pkg/storage ArticleStorage
+
+// ArticleStorage is an interface that defines the methods for interacting with the article storage.
+type ArticleStorage interface {
+	GetAll() ([]model.Article, error)
+	Create(article model.Article) (model.Article, error)
+	SaveAll(articles []model.Article) error
+	Delete(id int) error
+	GetByKeyword(keyword string) ([]model.Article, error)
```

Comment:
all these three funcs too specific, if you have more filter you will need to add APIs to the storage which is not good.
And this is a business logic actually, this layer should not care about it. 

so i suggest remove all of this from here and move it under the "service" or try to be abstract and be able to provide "filters" dynamically. But again I don't like word "filters" for the storage layer.